orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForCapacityMatchTask.groovy

```diff
@@ -43,6 +43,13 @@ class WaitForCapacityMatchTask extends AbstractInstancesCheckTask {
     if (!serverGroup.capacity || serverGroup.capacity.desired != instances.size()) {
       return false
     }
+
+    if (stage.context.get("capacity") && stage.context.get("capacity").get("desired")) {
```

Comment:
Why doesn't the above check work? Shouldn't the server group have the capacity set?

Reply:
This is possible my use case.  First, I have stage is set size to 0.  I think that it is timing during the resizing.  The above condition will satisfy and return true and it won't checking the size anymore and launch next stage.  So, some of the pod might not shutdown before remove the statefulset.

Reply:
@ajordens what are your thoughts on this change? I'm worried it'll have some weird consequences

Reply:
stage should have the desired instance.  return false will ask to recheck.  I can move the code the end before return true.  That might reduce some impact.  How do you think?

Reply:
I'm not sure what we're checking here ... it looks like this change is relying on groovy truth and offering up some different behavior when `capacity.desired == 0`?

I don't believe we should short-circuit this task until the actual instance count matches desired.

Is that problem that you don't want to wait after you have resized to zero?

Reply:
My test case is for statefulset and daemon controllers which don't need disable serverGroup.
- create a pipeline
- create resize server group stage and set to 0 ( currently running 4 pods)
- create a destroy server group stage to delete controller.

First onDemand call,
(!serverGroup.capacity || serverGroup.capacity.desired != instances.size())  ==> true (4=4)

Now, we reach the following line.  serverGroup.disabled = false.  Now, it will return true and stop request any update and wait.
!serverGroup.disabled ?
      WaitForUpInstancesTask.allInstancesMatch(stage, serverGroup, instances, interestingHealthProviderNames) :
      true

Reply:
My goal is make sure to wait.  If I delete controller before pods terminated.  It will have orphan pods.

Reply:
If you've resized to zero, `capacity.desired` would not equal `instances.count()` ... 0 != 4.

I must be missing something?